Kconfig.zephyr

```
@@ -442,6 +442,14 @@ config COMPILER_COLOR_DIAGNOSTICS
 	help
 	  Compiler diagnostic messages are colorized.
 
+config COMPILER_STATIC_ANALYSIS
+	bool "Compiler static analysis"
+	depends on "${ZEPHYR_TOOLCHAIN_VARIANT}" = "zephyr"
+	help
+	  Enables an static analysis of program flow and issues
+	  warnings for problems found. This feature only works
+	  on GCC, slows down the build and can produce false positives.
```

This is not the only option for doing static analysis as Zephyr also supports the use of stand-alone SCA tools, so as minimum we should refer reader there also.

```suggestion
	  Enables static analysis of program flow and issues
	  warnings for problems found during compilation.
	  This option does nothing on compilers which doesn't have
	  dedicated flags for static analysis during compilation.
	  See also Zephyr documentation on how to run dedicated
	  SCA Tools during Zephyr build using the ZEPHYR_SCA_VARIANT.
```